### DISCUSSION TITLE
Need Help with an unsupported device and "Herdsmen-converter" errors.

### DISCUSSION THREAD

**Discussion by chrishome7** [Koenkk/zigbee2mqtt#10886] [Devices]
I'm fairly new to this and I'm trying to to write a converter for an unsupported Linkind bulb ['ZBT-DIMLight-GLS0010']
model: 'ZL100010006'. 
Unfortunately I keep getting an error.

Error: Cannot find module 'zigbee-herdsman-converters/lib/extend'
Require stack:
- /app/lib/util/utils.js
- /app/lib/util/settings.js
- /app/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at /app/lib/util/externally-loaded.js:5:16
    at Script.runInContext (vm.js:143:18)
    at Script.runInNewContext (vm.js:148:17)
    at Object.runInNewContext (vm.js:303:38)
    at loadModuleFromText (/app/lib/util/utils.js:162:8)
    at loadModuleFromFile (/app/lib/util/utils.js:168:12)
 
I'm not sure what I'm doing incorrectly. I get the error testing other individuals successful converters.
 I'm running the current edge version on HA supervised.

**Comment by chrishome7** [comment]
I'm pretty sure its not an issue with my custom converter.
I've tried everything I can think of including dragging the entire herdsmen converter branch into the Z2M tree.

**Comment by Koenkk** [comment]
Could you post your external converter here?

**Reply by chrishome7** [reply]
Yes sir. This is what all I've gotten so far. I placed it under The Z2M tree (Custom_zigbee.js )

const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    zigbeeModel: ['ZBT-DIMLight-GLS0010'],
    model: 'ZL100010006',
    vendor: 'Linkind',
    description: 'Zigbee LED 9W 2700K A19 bulb, dimmable',
    extend: extend.light_onoff_brightness(),
};

module.exports = definition;

**Reply by chrishome7** [reply]
I've also tried it the flowing two ways.

const exposes = require('zigbee-herdsman-converters/lib/exposes');
const fz = {...require('zigbee-herdsman-converters/converters/fromZigbee'), legacy: require('zigbee-herdsman-converters/lib/legacy').fromZigbee};
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

module.exports = [
    {
      zigbeeModel: ['ZBT-DIMLight-GLS0010'],
    model: 'ZL100010006',
    vendor: 'Linkind',
    description: 'Zigbee LED 9W 2700K A19 bulb, dimmable',
    extend: extend.light_onoff_brightness(),
    },    
];

And with the header structured like soo

const exposes = require('../lib/exposes');
const fz = {...require('../converters/fromZigbee'), legacy: require('../lib/legacy').fromZigbee};
const reporting = require('../lib/reporting');
const extend = require('../lib/extend');
const tz = require('../converters/toZigbee');
const e = exposes.presets;
const ea = exposes.access;

**Reply by chrishome7** [reply]
I also tried adding the entire Herdsmen Branch under the Z2M Dir as others had suggested.
![Z2M_1capture](https://user-images.githubusercontent.com/65036140/150234139-774bfe14-166c-4688-9e1f-58ad7d5db572.jpg)

**Reply by chrishome7** [reply]
Turns out The bulb in question is already supported. I was unaware of the repo change and my HA instance wasn't reporting as being out of date. I will try a sample connector with the new updated version. 
I apologize. Thanks for your time and assistance.
